tests/test_minimal/test_tools/test_expand_paths.py

```diff
@@ -28,38 +28,37 @@ def test_only_folder_match(tmpdir):
     file2.touch()
 
     # Add another sub-nested folder with a folder
-    sub_directory3 = sub_directory1 / "a_simple_pattern_3"
-    sub_directory3.mkdir(exist_ok=True)
+    sub_directory3 = sub_directory1 / "intermediate_nested" / "a_simple_pattern_3"
+    sub_directory3.mkdir(exist_ok=True, parents=True)
     file3 = sub_directory3 / "a_simple_pattern_3.bin"
     file3.touch()
 
     # Specify source data (note this assumes the files are arranged in the same way as in the example data)
     source_data_spec = {
         "a_source": {
             "base_directory": base_directory,
-            "folder_path": "a_simple_pattern_{session_id}",
+            "folder_path": "{subject_id}/a_simple_pattern_{session_id}",
         }
     }
 
-    # Instantiate LocalPathExpander
-
     path_expander = LocalPathExpander()
     metadata_list = path_expander.expand_paths(source_data_spec)
     folder_paths = [metadata_match["source_data"]["a_source"]["folder_path"] for metadata_match in metadata_list]
 
-    expected = {str(sub_directory1), str(sub_directory2), str(sub_directory3)}
+    # Note that sub_directory3 is not included because it does not conform to the pattern
+    expected = {str(sub_directory1), str(sub_directory2)}
```

Comment:
can you also test that the metadata was extracted correctly?

Reply:
Done. I also added a docstring to `extract_metadata`